src/thunderbots/software/ai/hl/stp/evaluation/calc_best_shot.h

```diff
@@ -0,0 +1,97 @@
+#ifndef PROJECT_CALC_BEST_SHOT_H
+#define PROJECT_CALC_BEST_SHOT_H
+
+#include "ai/world/field.h"
+#include "ai/world/robot.h"
+#include "ai/world/world.h"
+#include "geom/point.h"
+
+namespace Evaluation
+{
+    /**
+     * Finds the length of the largest continuous interval (angle-wise) of the
+     * goal that can be seen from a point.
+     *
+     * @param f-field is needed to calculate the location of the goalposts based on field
+     * length and goal width
+     *
+     * @param obstacles-obstacles is a list of all the obstacles in the way between the
+     * point and the net
+     *
+     * @param p-the point that the shot is being calculated from
+     *
+     * @param radius-the multiplier to the radius of the robot,
+     * you can decrease the multiplier to make it easier to shoot.
+     *
+     * @return the point and the score (angle),
+     * where the score will be 0 if the point is invalid.
+     */
+    std::pair<Point, Angle> calcBestShot(const Field& f,
```

Comment:
Since all these functions are calulating the best shot on the enemy goal, can we rename all these functions to `calcBestShotOnEnemyGoal` of something similar? Just to be more explicit about where the shot is.

Reply:
Renamed all functions